File: cpp/generated_utils/src/test-run-single-helpers.cpp
#include "/home/runner/work/elizaos-cpp/elizaos-cpp/SWEagent/tests/test-run-single-helpers.h"

std::shared_ptr<Promise<void>> MockRuntime::createSession(std::shared_ptr<CreateBashSessionRequest> _request)
{
    return std::shared_ptr<Promise<void>>();
}

std::shared_ptr<Promise<std::shared_ptr<CommandResult>>> MockRuntime::execute(std::shared_ptr<Command> _command)
{
    return object{
        object::pair{std::string("exitCode"), 0}, 
        object::pair{std::string("stdout"), string_empty}, 
        object::pair{std::string("stderr"), string_empty}
    };
}

std::shared_ptr<Promise<std::shared_ptr<ReadFileResponse>>> MockRuntime::readFile(std::shared_ptr<ReadFileRequest> request)
{
    auto content = OR((this->files->get(request->path)), (string_empty));
    return object{
        object::pair{std::string("content"), std::string("content")}
    };
}

std::shared_ptr<Promise<void>> MockRuntime::writeFile(std::shared_ptr<WriteFileRequest> request)
{
    this->files->set(request->path, request->content);
    return std::shared_ptr<Promise<void>>();
}

std::shared_ptr<Promise<void>> MockRuntime::upload(std::shared_ptr<UploadRequest> _request)
{
    return std::shared_ptr<Promise<void>>();
}

MockDeployment::MockDeployment() : AbstractDeployment() {
    this->runtime = std::make_shared<MockRuntime>();
}

std::shared_ptr<Promise<void>> MockDeployment::start()
{
    return std::shared_ptr<Promise<void>>();
}

std::shared_ptr<Promise<void>> MockDeployment::stop()
{
    return std::shared_ptr<Promise<void>>();
}

File: SWEagent/tests/test-run-single-helpers.ts
/**
 * Helper classes for test-run-single.test.ts
 */

import { 
  AbstractDeployment,
  AbstractRuntime,
  BashAction,
  BashActionResult,
  BashInterruptAction,
  Command,
  CommandResult,
  CreateBashSessionRequest,
  ReadFileRequest,
  ReadFileResponse,
  WriteFileRequest,
  UploadRequest,
} from '../src/environment/deployment';

/**
 * Mock runtime for testing
 */
export class MockRuntime extends AbstractRuntime {
  private files: Map<string, string> = new Map();
  private sessionOutput: string = '';

  async createSession(_request: CreateBashSessionRequest): Promise<void> {
    // Mock implementation
  }

  async runInSession(action: BashAction | BashInterruptAction): Promise<BashActionResult> {
    if ('type' in action && action.type === 'interrupt') {
      return { output: '', exitCode: 0 };
    }

    const bashAction = action as BashAction;

    // Mock some basic commands
    if (bashAction.command.startsWith('echo ')) {
      const text = bashAction.command.substring(5).replace(/['"]/g, '');
      return { output: text + '\n', exitCode: 0 };
    }

    if (bashAction.command === 'ls') {
      return { output: 'file1\nfile2\n', exitCode: 0 };
    }

    if (bashAction.command.startsWith('sleep ')) {
      const seconds = parseFloat(bashAction.command.substring(6));
      if (bashAction.timeout && bashAction.timeout < seconds) {
        throw new Error('Command timeout');
      }
      return { output: '', exitCode: 0 };
    }

    return { output: this.sessionOutput, exitCode: 0 };
  }

  async execute(_command: Command): Promise<CommandResult> {
    return { exitCode: 0, stdout: '', stderr: '' };
  }

  async readFile(request: ReadFileRequest): Promise<ReadFileResponse> {
    const content = this.files.get(request.path) || '';
    return { content };
  }

  async writeFile(request: WriteFileRequest): Promise<void> {
    this.files.set(request.path, request.content);
  }

  async upload(_request: UploadRequest): Promise<void> {
    // Mock implementation
  }
}

/**
 * Mock deployment for testing
 */
export class MockDeployment extends AbstractDeployment {
  runtime: MockRuntime;

  constructor() {
    super();
    this.runtime = new MockRuntime();
  }

  async start(): Promise<void> {
    // Mock implementation
  }

  async stop(): Promise<void> {
    // Mock implementation
  }
}
